relecov_tools/json_validation.py

```diff
@@ -101,6 +103,43 @@ def __init__(
         else:
             self.excel_sheet = excel_sheet
 
+        # Parse file containing sample IDs registry
+        if registry is not None and relecov_tools.utils.file_exists(registry):
+            self.registry_path = registry
+        else:
+            config_json = ConfigJson(extra_config=True)
+            try:
+                default_path = config_json.get_topic_data(
+                    "validate_config", "default_sample_id_registry"
+                )
+            except KeyError:
+                default_path = None
+
+            if default_path and relecov_tools.utils.file_exists(default_path):
+                self.registry_path = default_path
+            else:
+                stderr.print(
+                    "[yellow]No valid ID registry found. Please select the file manually."
+                )
+                prompted_path = relecov_tools.utils.prompt_path(
+                    "Select the JSON file with registered unique sample IDs"
+                )
+                if relecov_tools.utils.file_exists(prompted_path):
+                    self.registry_path = prompted_path
+                else:
+                    stderr.print(
+                        "[red]No valid ID registry file could be found or selected."
+                    )
+                    sys.exit(1)
+
+        # Read id registry file
+        try:
+            self.id_registry = relecov_tools.utils.read_json_file(self.registry_path)
+        except Exception as e:
+            stderr.print(f"[red]Failed to read ID registry JSON: {e}")
+            self.log.error(f"Failed to read ID registry JSON: {e}")
+            sys.exit(1)
```

Comment:
I'd raise the error here instead as no user interaction is needed. (Not the same for registry_path as you might need user interaction there)
```suggestion
            raise
```